apis/config/v1beta1/configuration_types.go

```diff
@@ -67,6 +67,11 @@ type Configuration struct {
 	// QueueVisibility is configuration to expose the information about the top
 	// pending workloads.
 	QueueVisibility *QueueVisibility `json:"queueVisibility,omitempty"`
+
+	// SingleInstanceAdmissionCheckControllers contains a list of AdmissionCheck controller names for which
+	// at most one AdmissionCheck managed by it can be configured in a ClusterQueue.
+	// A ClusterQueue that violates this rule is marked as Inactive.
+	SingleInstanceAdmissionCheckControllers []string `json:"singleInstanceAdmissionCheckControllers,omitempty"`
```

Comment:
Why do we need this? Can we just verify that there is at most one MultiKueue admission check per ClusterQueue? Alternatively, (stronger) if there is at least one MultiKueue admission check, there are no other admission checks for the ClusterQueue. 

It feels not many users will fully understand how to set this field, and might be afraid it can block some of their use cases.

If we need a new API field I think we should update the KEP. WDYT @mwielgus ?

Reply:
There is no knowledge of MultiKueue and is should not be at ClusterQueue level. Also the need of having at most one admission check managed by a specific admission check controller in a queue is an attribute of that controller. Currently we don't have an API object for the Admission Check Controllers, adding one might be to much for what we need now and will add an extra configuration step for the average kueue user.

Reply:
I think if this change impacts the API it should be discussed in KEP. 

The other potential options would be:
1. keep the mechanism, but hard-code the multi-kueue controller to require single instance (no API change so we could move on without KEP, I don't see much value asking the user to configure it, leaving room for error)
2. if needed, should this not be the configuration on AdmissionCheck, rather than global?

WDYT @alculquicondor ?

Reply:
> 2\. if needed, should this not be the configuration on AdmissionCheck, rather than global?
It need to have the same value for all the AdmissionChecks managed by a specific controller.

Reply:
> It need to have the same value for all the AdmissionChecks managed by a specific controller.

Makes sense, still non-obvious to me we prefer to expose this as a user-facing configuration.

I would like to hear form @alculquicondor or @mwielgus if we proceed with this first since it is not part of the KEP currently.

Reply:
This should not be configurable. This is an inherit limitation of MultiKueue.

Then, it should be the MultiKueue controller that should somehow detect the problem and inform the ClusterQueue that there is a problem.

But let's leave this discussion for the next release.

Reply:
To clarify, what I mean by `But let's leave this discussion for the next release` is that we can leave this PR for the next release.

If you want to merge it in this release, then you should address the fact that this should NOT be configurable.

Reply:
/hold

Reply:
Let's control it from the admission check controller, via a new AC Status field.
/unhold

Reply:
Thanks!